covalent/_workflow/electron.py

```diff
@@ -383,13 +434,27 @@ def __call__(self, *args, **kwargs) -> Union[Any, "Electron"]:
 
             return bound_electron
 
-        # Add a node (other than sublattices) to the transport graph of the active lattice
-        self.node_id = active_lattice.transport_graph.add_node(
-            name=self.function.__name__,
-            function=self.function,
-            metadata=self.metadata.copy(),
-            function_string=get_serialized_function_str(self.function),
-        )
+        # Add a node to the transport graph of the active lattice. Electrons bound to nodes will never be packed with the
+        # 'master' Electron.
+        if self.packing_tasks:
+            # Add non-sublattice node to the transport graph of the active lattice.
+            self.node_id = active_lattice.transport_graph.add_node(
+                name=self.function.__name__,
+                function=self.function,
+                metadata=self.metadata.copy(),
+                function_string=get_serialized_function_str(self.function),
+                task_group_id=self.task_group_id,
+            )
+
+        else:
+            # Default to node_id as the task_group_id.
+            self.node_id = active_lattice.transport_graph.add_node(
+                name=self.function.__name__,
+                function=self.function,
+                metadata=self.metadata.copy(),
+                function_string=get_serialized_function_str(self.function),
+            )
+            self.task_group_id = self.node_id
```

Comment:
It seems that the only difference in these if/else conditions is that in the else clause there is a `task_group_id`. To prevent code duplication can't we just do:
```python
....
task_group_id = self.task_group_id if self.packing_tasks else None
....
```

Reply:
That's a good point for `add_node`. The other difference is that only if `packing_tasks` is `False` should `self.task_group_id` be updated with the `node_id` of the newly added node; otherwise it should remain at the smallest `node_id` in the task group.